Kev/3 api key

junhaow1 opened the pull request:
## Pull Request Details

Link to associated issue: https://github.com/anusii/moviestar/issues/3

# Checklist
Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (make prep or flutter analyze lib)
- [x] Tested on at least one device
  - [x] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

Commit:
Renamed the ProfileScreen class to SettingsScreen

Commit:
Add a secure storage service for the API key
Add an API key input field to the settings screen
Modify the movie service to use the stored key

Commit:
Renamed my_list_screen.dart to to_watch_screen.dart
Updated the screen class from MyListScreen to ToWatchScreen
Changed the screen title from "My List" to "To Watch"
Updated the settings screen to use "To Watch" and "Watched" instead of "My List" and "Watch History"
Updated imports and references accordingly

Commit:
Merge branch 'kev/1_initial_ideas' into kev/3_api_key

Commit:
Changed apiKeyService.apiKey to apiKeyService.getApiKey()

Commit:
Merge branch 'kev/3_api_key' into kev/4_rename_list

Commit:
fix duplicate items

Commit:
Merge branch 'kev/1_initial_ideas' into kev/3_api_key

Commit:
fix conflict

junhaow1 committed:
Merge pull request #2 from anusii/kev/1_initial_ideas

Kev/1 initial ideas , app

Commit:
Merge branch 'dev' into kev/3_api_key

Commit:
fix conflict

Commit:
Merge branch 'kev/3_api_key' into kev/4_rename_list

anushkavidanage commented:
Even after entering the new API key, I get the network exception error.

<img width="633" alt="error" src="https://github.com/user-attachments/assets/fbe035ad-4c1f-44ab-b2a4-1fec701896f0" />

junhaow1 committed:
run on macos

junhaow1 committed:
Merge pull request #22 from anusii/kev/4_rename_list

Kev/4 rename list

junhaow1 committed:
add gitignore

junhaow1 committed:
Added a clickable link to TMDB's website where users can get an API key
Added a helpful description explaining that the API key is required for fetching movie data
Added the url_launcher package and implemented the URL launching functionality

junhaow1 commented:
> > Even after entering the new API key, I get the network exception error.
> > <img alt="error" width="633" src="https://private-user-images.githubusercontent.com/81279146/446316846-fbe035ad-4c1f-44ab-b2a4-1fec701896f0.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc4NzM3NjMsIm5iZiI6MTc0Nzg3MzQ2MywicGF0aCI6Ii84MTI3OTE0Ni80NDYzMTY4NDYtZmJlMDM1YWQtNGMxZi00NGFiLWIyYTQtMWZlYzcwMTg5NmYwLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTIyVDAwMjQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyYjg5OTRiY2M5NzQ0MDJhZjdlNjY5NDY0ZDE5ZjkzNmU3MjU5YTJkZWQ5YTM5NDcyN2ZlMDU3Y2Q5NDU5YjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-UWnjFhho-_R36pcqGSa7Hh5_Btb5Kgd8kNrGX5GPAI">
> 
> OK, I had to login again to reset the home page. A couple of suggestions are listed below.
> 
> * [ ]  After entering the new API key, load the movie data without having to login again
> * [ ]  Upon login in for the first time, check if the user has an API key set up and if it is not setup direct the user to the API key input page with a message asking to enter the API key
> * [x]  On the API key input page, include the TMDB link as an example to let the user know where can they obtain the API key.

<img width="403" alt="image" src="https://github.com/user-attachments/assets/c622571d-86ef-4ea1-83a7-8f6e1e4738df" />

junhaow1 commented:
<img width="1242" alt="image" src="https://github.com/user-attachments/assets/3169d743-d1fd-402c-b037-f94bb3bbae86" />

junhaow1 committed:
implemented all the necessary changes to ensure that when we enter a new API key, the movie data will load immediately without requiring us to log in again or restart the app

junhaow1 committed:
add git ingore macos folder

junhaow1 committed:
ignore

anushkavidanage reviewed (approved):
Looks good. Thanks Kevin

cdawei commented:
The app worked well, the only issue I got is the following exception after I edited my API key and clicked the `Save API Key` button.
```
The following LateError was thrown while dispatching notifications for ApiKeyService:
LateInitializationError: Field '_screens@37043735' has already been initialized.

When the exception was thrown, this was the stack:
#0      LateError._throwFieldAlreadyInitialized (dart:_internal-patch/internal_patch.dart:209:5)
#1      _MyHomePageState._screens= (package:moviestar/main.dart)
#2      _MyHomePageState._buildScreens (package:moviestar/main.dart:147:5)
#3      _MyHomePageState._onApiKeyChanged.<anonymous closure> (package:moviestar/main.dart:136:7)
#4      State.setState (package:flutter/src/widgets/framework.dart:1207:30)
#5      _MyHomePageState._onApiKeyChanged (package:moviestar/main.dart:135:5)
#6      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:439:24)
#7      ApiKeyService.setApiKey (package:moviestar/services/api_key_service.dart:16:5)
<asynchronous suspension>
#8      _SettingsScreenState.build.<anonymous closure> (package:moviestar/screens/settings_screen.dart:216:23)
<asynchronous suspension>

The ApiKeyService sending notification was:
  Instance of 'ApiKeyService'
```

Commit:
Removed the attempt to rebuild the screens list in _onApiKeyChanged()

junhaow1 commented:
> The app worked well, the only issue I got is the following exception after I edited my API key and clicked the `Save API Key` button.

fixed
